src/lib_json/json_reader.cpp

```diff
@@ -1051,12 +1051,9 @@ bool OurReader::parse(const char* beginDoc,
   nodes_.pop();
   Token token;
   skipCommentTokens(token);
-  if (features_.failIfExtra_) {
-    if ((features_.strictRoot_ || token.type_ != tokenError) &&
-        token.type_ != tokenEndOfStream) {
-      addError("Extra non-whitespace after JSON value.", token);
-      return false;
-    }
+  if (features_.failIfExtra_ && (token.type_ != tokenEndOfStream)) {
```

Comment:
Where did the check for strict root and token type go? Don't we need all the checks still?

Reply:
`(features_.strictRoot_ || token.type_ != tokenError)` caused certain inputs to be unexpectedly accepted if `failIfExtra_` is on and `strictRoot_` is off. For example, take the input `123 foo`: 123 is accepted, but `foo` produces a tokenError. So there was no failure, despite "extra" data being present. The original PR #1001 explains the problem.

Generally, `strictRoot_` is also checked in a later (beyond the diff), so that's still honored.

If you think there could be some input that's handles incorrectly after this change I can check whether the unit tests cover that (and extend those if necessary).